dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp

```diff
@@ -469,6 +472,42 @@ void DAGStorageInterpreter::executeImpl(DAGPipeline & pipeline)
     }
 }
 
+// TODO:Question: here we assume that, if the columns' id and data type in query is the same as the columns in TiDB,
+// we think we can directly do read, and don't need sync schema.
+// compare the columns in table_scan with the columns in storages, to check if the current schema is satisified this query.
+// column.name are always empty from table_scan, and column name is not necessary in read process, so we don't need compare the name here.
+bool compareColumns(const TiDBTableScan & table_scan, const DM::ColumnDefines & cur_columns, const LoggerPtr & log)
+{
+    const auto & columns = table_scan.getColumns();
+    std::unordered_map<ColumnID, DM::ColumnDefine> column_id_map;
+    for (const auto & column : cur_columns)
+    {
+        column_id_map[column.id] = column;
+    }
+
+    for (const auto & column : columns)
+    {
+        // Exclude virtual columns, including EXTRA_HANDLE_COLUMN_ID, VERSION_COLUMN_ID,TAG_COLUMN_ID,EXTRA_TABLE_ID_COLUMN_ID
+        if (column.id < 0)
+        {
+            continue;
+        }
+        auto iter = column_id_map.find(column.id);
+        if (iter == column_id_map.end())
+        {
+            LOG_WARNING(log, "the column with id {} in query is not found in current columns", column.id);
+            return false;
+        }
```

Comment:
Have you checked that will generated column makes tiflash always sync schema?
similar problem caused by generated column: https://github.com/pingcap/tiflash/issues/7468

Reply:
For the table with generated column, the generated column has its own column id in table_info , just like a normal column. So we don't need sync schema when read the table with generated columns.